influxdb_client/client/write/retry.py

```diff
@@ -54,12 +76,26 @@ def get_backoff_time(self):
         if consecutive_errors_len < 0:
             return 0
 
-        # Full Jitter strategy
-        backoff_value = self.backoff_factor * (self.exponential_base ** consecutive_errors_len) * self._random()
-        return min(self.max_retry_delay, backoff_value)
+        delay_range = self.backoff_factor
+        i = 1
+        while i <= consecutive_errors_len:
+            i += 1
+            delay_range = delay_range * self.exponential_base
+            if delay_range > self.max_retry_delay:
+                break
+
+        delay = delay_range * self._random()
+        # at least min_retry_delay
+        delay = max(self.min_retry_delay, delay)
+        # at most max_retry_delay
+        delay = min(self.max_retry_delay, delay)
+        return delay
 
     def increment(self, method=None, url=None, response=None, error=None, _pool=None, _stacktrace=None):
         """Return a new Retry object with incremented retry counters."""
+        if self.retry_timeout < datetime.now():
```

Comment:
Does it also react the same way when retry is disabled?

Reply:
yes, max_retry_time=0 means retry is disabled, here is the test:
https://github.com/influxdata/influxdb-client-python/blob/5b16095336677cd00c0e8d5c2f1aec254c53dbdf/tests/test_WriteApiBatching.py#L257